packages/admin-ui/src/filters/filter-disclosure.tsx

```diff
@@ -2,24 +2,37 @@ import type { ReactNode } from 'react'
 import React from 'react'
 import { Button } from '../button'
 import { IconCaretUp } from '@vtex/phosphor-icons'
-import type { MenuState } from 'ariakit'
 import { MenuButton } from 'ariakit'
 
 import * as style from './filter.style'
+import type { GenericFilterStateReturn } from '.'
+import { AppliedListLabel } from './filter-applied-list-label'
+import type { AnyObject } from '..'
 
 export const FilterDisclosure = (props: FilterDisclosureProps) => {
-  const { state, children, id } = props
+  const { state, children, appliedItems } = props
+
+  const disclosureId = state.baseId ? `${state.baseId}-disclosure` : undefined
 
   return (
-    <Button as={MenuButton as any} state={state} csx={style.disclosure} id={id}>
+    <Button
+      as={MenuButton as any}
```

Comment:
Maybe you could use the MenuButton and just pass to it our button styles. What do you think? Make sense in this case?

Reply:
Might make sense, I'm still thinking about a better way to handle these "as" and will come back to it and other ariakit topics in the future